timezone/src/modules/Timezone.rb

```diff
@@ -245,6 +251,12 @@ def get_zonemap
     # @return	the number of the region that contains the timezone
     #
     def Set(zone, really)
+      # Do not update the timezone if it's forced and it was already set
+      if readonly && !@timezone.empty?
+        log.info "Timezone is read-only and cannot be changed"
+        return -1 # Not sure about this
```

Comment:
I'm returning a `-1` to avoid changing the method's signature as some code could rely on this.